compiler/rustc_codegen_ssa/src/mir/debuginfo.rs

```diff
@@ -341,11 +341,24 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                             && var.source_info.scope == mir::OUTERMOST_SOURCE_SCOPE
                         {
                             let arg_index = place.local.index() - 1;
-
-                            // FIXME(eddyb) shouldn't `ArgumentVariable` indices be
-                            // offset in closures to account for the hidden environment?
-                            // Also, is this `+ 1` needed at all?
-                            VariableKind::ArgumentVariable(arg_index + 1)
+                            // Rust compiler decomposes every &str or slice argument into two components
+                            // a pointer to the memory address storing the data and an u32 representing
+                            // the length of the str or slice. These components were then used to reconstruct
+                            // the original argument inside the body of the function. Since the original argument
+                            // was declared inside the body of the function rather than being passed by the function's
```

Comment:
Can you add a pointer to the code which creates the "debug spill" local variable, so that this behavior is more understandable for the reader?